stem/model.py

```diff
@@ -1,24 +1,164 @@
         self.body_model_parts: List[BodyModelPart] = []
         self.process_model_parts: List[ModelPart] = []
 
+        self.extrusion_length: Optional[Sequence[float]] = None
+
+    def __del__(self):
+        """
+        Destructor of the Model class. Finalizes the gmsh_io instance.
+
+        """
+        self.gmsh_io.finalize_gmsh()
+
+    def __get_geometry_from_geo_data(self, geo_data: Dict[str, Any]):
+        """
+        Get the geometry from the geo_data as generated by gmsh_io.
+
+        Args:
+            - geo_data (Dict[str, Any]): dictionary containing the geometry data as generated by the gmsh_io
+
+        """
+
+        self.geometry = Geometry.create_geometry_from_geo_data(geo_data)
+
+    def add_all_layers_from_geo_file(self, geo_file_name: str, body_names: Sequence[str]):
+        """
+        Add all physical groups from a geo file to the model. The physical groups with the names in body_names are
+        added as body model parts, the other physical groups are added as process model parts.
+
+        Args:
+            - geo_file_name (str): name of the geo file
+            - body_names (Sequence[str]): names of the physical groups which should be added as body model parts
+
+        """
+
+        # read the geo file and generate the geo_data dictionary
+        self.gmsh_io.read_gmsh_geo(geo_file_name)
+
+        # Reset the gmsh instance with the geo data, as read from the geo file
+        self.gmsh_io.generate_geo_from_geo_data()
+
+        geo_data = self.gmsh_io.geo_data
+
+        # Create geometry and model part for each physical group in the gmsh geo_data
+        model_part: Union[ModelPart, BodyModelPart]
+        for group_name in geo_data["physical_groups"].keys():
+
+            # create model part, if the group name is in the body names, create a body model part, otherwise a process
+            # model part
+            if group_name in body_names:
+                model_part = BodyModelPart()
+            else:
+                model_part = ModelPart()
+
+            # set the name and geometry of the model part
+            model_part.name = group_name
+            model_part.get_geometry_from_geo_data(geo_data, group_name)
+
+            # add model part to either body model parts or process model part
+            if isinstance(model_part, BodyModelPart):
+                self.body_model_parts.append(model_part)
+            else:
+                self.process_model_parts.append(model_part)
+
+    def add_soil_layer_by_coordinates(self, coordinates: Sequence[Sequence[float]],
+                       material_parameters: Union[SoilMaterial, StructuralMaterial], name: str,
+                       ):
+        """
+        Adds a soil layer to the model by giving a sequence of 2D coordinates. In 3D the 2D geometry is extruded in
+        the direction of the extrusion_length
+
+        Args:
+            - coordinates (Sequence[Sequence[float]]): The coordinates of the soil layer.
```

Comment:
- coordinates (Sequence[Sequence[float]]): The 2D coordinates of the soil layer.